rskj-core/src/main/java/co/rsk/net/light/LightProcessor.java

```diff
@@ -0,0 +1,222 @@
+    }
+
+    public void processBlockReceiptsMessage(long id, List<TransactionReceipt> blockReceipts, LightPeer lightPeer) {
+        throw new UnsupportedOperationException("Not supported BlockReceipt processing");
+    }
+
+    public void processGetTransactionIndex(long id, byte[] hash, LightPeer lightPeer) {
+        logger.debug("Get Transaction Index Message Received");
+
+        TransactionInfo txinfo = blockchain.getTransactionInfo(hash);
+
+        if (txinfo == null) {
+            // Don't waste time sending an empty response.
+            return;
+        }
+
+        byte[] blockHash = txinfo.getBlockHash();
+        long blockNumber = blockchain.getBlockByHash(blockHash).getNumber();
+        long txIndex = txinfo.getIndex();
+
+        TransactionIndexMessage response = new TransactionIndexMessage(id, blockNumber, blockHash, txIndex);
+        lightPeer.sendMessage(response);
+    }
+
+    public void processTransactionIndexMessage(long id, long blockNumber, byte[] blockHash, long txIndex, LightPeer lightPeer) {
+        throw new UnsupportedOperationException("Not supported TransactionIndexMessage processing");
+    }
+
+    public void processGetCodeMessage(long requestId, byte[] blockHash, byte[] address, LightPeer lightPeer) {
+        String blockHashLog = Hex.toHexString(blockHash);
+        String addressLog = Hex.toHexString(address);
+        logger.trace("Processing code request {} block {} code {}", requestId, blockHashLog, addressLog);
+
+        final Block block = blockStore.getBlockByHash(blockHash);
+
+        if (block == null) {
+            // Don't waste time sending an empty response.
+            return;
+        }
+
+        RepositorySnapshot repositorySnapshot = repositoryLocator.snapshotAt(block.getHeader());
+        RskAddress addr = new RskAddress(address);
+        Keccak256 codeHash = repositorySnapshot.getCodeHash(addr);
+
+        CodeMessage response = new CodeMessage(requestId, codeHash.getBytes());
+        lightPeer.sendMessage(response);
+    }
+
+    public void processCodeMessage(long id, byte[] codeHash, LightPeer lightPeer) {
+        throw new UnsupportedOperationException("Not supported Code processing");
+    }
+
+    public void processGetAccountsMessage(long id, byte[] blockHash, byte[] addressHash, LightPeer lightPeer) {
+        logger.debug("Get Accounts Message Received: id {}, blockhash: {}, addressHash {}", id, blockHash, addressHash);
+
+        final Block block = blockStore.getBlockByHash(blockHash);
+
+        if (block == null) {
+            // Don't waste time sending an empty response.
+            return;
+        }
+
+        RepositorySnapshot repositorySnapshot = repositoryLocator.snapshotAt(block.getHeader());
+        RskAddress address = new RskAddress(addressHash);
+        AccountState state = repositorySnapshot.getAccountState(address);
+
+        AccountsMessage response = new AccountsMessage(id, new byte[] {0x00}, state.getNonce().longValue(),
```

Comment:
I would add a TODO comment to mark that the merkle proof is yet to be implemented, and the same below